extensions/2.0/Khronos/KHR_texture_procedurals/README.md

```diff
@@ -0,0 +1,1351 @@
+* To a `node input` from a node `output` by specifying a `node` value which is an index into the graph's `nodes` array.
+* To a nodegraph `output` from a node `output` by specifying a `node` value which is an index into the graph's `nodes` array.
+
+If the upstream node has multiple outputs, then an `output` value which is an index into the the upstream nodes `outputs` array  __must__ additionally be specified. 
+
+
+The following example shows all the combinations of connections are shown. The sample logic adds two colors together and outputs the result. 
+
+* One color is exposed as a graph input (`graph_in`), while the other is internal to the graph as a constant node (`constant1`). 
+* The add node (`add1`) has two input ports which take as input the result of `graph_in` and `constant1`.
+* The output node on the graph (`graph_out`) is used to route the output value from the add node.
+
+Note that the output ports are specified even if the upstream node only has one output port.
+
+<details>
+<summary>Example</summary>
+
+```JSON
+{
+  "procedurals" [
+    {
+      "name": "nodegraph1",
+      "nodetype": "nodegraph",
+      "type": "color3",
+      "inputs": [
+        {
+          "name": "graph_in",
+          "nodetype": "input",
+          "type": "color3",
+          "value": [ 1, 0, 0 ]
+        }
+      ],
+      "nodes" [
+          {
+            "name": "constant1",
+            "nodetype": "constant",
+            "type": "color3",
+            "value": [ 1, 1, 0 ],
+            "outputs": [
+              {
+                "name": "out",
+                "nodetype": "output",
+                "type": "color3"
+              }
+            ]
+          },
+          {
+            "name": "add1",
+            "nodetype": "add",
+            "type": "color3",
+            "inputs": [
+              {
+                "name": "in1",
+                "nodetype": "input",
+                "type": "color3",
+                "value": [ 1, 0, 0 ]
+              },
+              {
+                "name": "in2",
+                "nodetype": "input",
+                "type": "color3",
+                "node": 0,
+                "output": 0
+              }
+            ],
+            "outputs": [
+              {
+                "name": "out",
+                "nodetype": "output",
+                "type": "color3"                
+              }
+            ]
+          }
+        ],
+
+        "outputs" [
+          {
+            "name": "graph_out",
+            "nodetype": "output",
+            "type": "color3",
+            "node": 1,
+            "output": 0
+          },
+        ]
+    }
+}
+```
+
+</details>
+
+### Material Binding
+
+To connect a graph `output` to a surface or displacement shader input the procedural extension can be declared within the a texture reference for a given material in the `materials` array.
```

Comment:
I propose to simply connect the procedurals to an image in the `images`array - if this image is created at load time or dynamically created at runtime is up to the implementation.